packages/buttons/src/styled/StyledIcon.ts

```diff
@@ -5,16 +5,34 @@
  * found at http://www.apache.org/licenses/LICENSE-2.0.
  */
 
-import styled from 'styled-components';
+import styled, { css, ThemeProps, DefaultTheme } from 'styled-components';
 import React, { Children } from 'react';
 import { DEFAULT_THEME, retrieveComponentStyles } from '@zendeskgarden/react-theming';
 
 const COMPONENT_ID = 'buttons.icon';
 
 interface IStyledIconProps {
   isRotated: boolean;
+  position?: 'start' | 'end';
 }
 
+const sizeStyles = (props: IStyledIconProps & ThemeProps<DefaultTheme>) => {
+  let marginProperty;
+
+  if (props.position === 'start') {
+    marginProperty = `margin-${props.theme.rtl ? 'left' : 'right'}`;
+  } else if (props.position === 'end') {
+    marginProperty = `margin-${props.theme.rtl ? 'right' : 'left'}`;
+  }
```

Comment:
Is it worthwhile for the `start/end` logic to be handled at the `<Button>` level (maybe with :first-of-type or something)? This may drop the requirement for a `position` prop and allow a single `Button.Icon` accessor or simplified API.

Reply:
If you look at the underlying DOM structure, there's not a way to reliably target the difference between:
- start icon
- end icon
- normal button icon

That's why I'm using the prop to help here.

Reply:
You probably already noted, but the `position` prop is effectively private – only exists at the `StyledIcon` level and not exposed in the public element API.